spec/unit/autoyast_spec.rb

```diff
@@ -136,4 +145,24 @@
       expect(autoyast.export_name).to eq("description-autoyast")
     end
   end
+
+  describe "#outgoing_ip" do
+    let(:autoyast) { Autoyast.new(description) }
+    let(:ip_route_ls) {
+<<-EOF
+default via 10.100.255.254 dev em1
+10.100.0.0/16 dev em1  proto kernel  scope link  src 10.100.2.35
+172.17.0.0/16 dev docker0  proto kernel  scope link  src 172.17.42.1
+192.168.121.0/24 dev virbr1  proto kernel  scope link  src 192.168.121.1
+192.168.122.0/24 dev virbr0  proto kernel  scope link  src 192.168.122.1
+EOF
+    }
+
+    it "returns the current outgoing ip" do
+      expect(Cheetah).to receive(:run).with(
+        "ip", "route", "ls", stdout: :capture
+      ).and_return(ip_route_ls)
+      expect(autoyast.outgoing_ip).to eq("10.100.2.35")
+    end
+  end
```

Comment:
Can you please add a test which handles the situation if no default route is set?